qcodes/instrument/base.py

```diff
@@ -12,7 +12,246 @@
 from .function import Function
 
 
-class Instrument(Metadatable, DelegateAttributes):
+class InstrumentBase(Metadatable, DelegateAttributes):
+    """
+    Base class for all QCodes instruments and instrument channels
+
+    Args:
+        name (str): an identifier for this instrument, particularly for
+            attaching it to a Station.
+
+        metadata (Optional[Dict]): additional static metadata to add to this
+            instrument's JSON snapshot.
+
+
+    Attributes:
+        name (str): an identifier for this instrument, particularly for
+            attaching it to a Station.
+
+        parameters (Dict[Parameter]): All the parameters supported by this
+            instrument. Usually populated via ``add_parameter``
+
+        functions (Dict[Function]): All the functions supported by this
+            instrument. Usually populated via ``add_function``
+    """
+    def __init__(self, name, **kwargs):
+        self.name = str(name)
+        self.parameters = {}
+        self.functions = {}
+        self.submodules = {}
```

Comment:
Why does the `InstrumentBase` have submodules? Does it need them? And if so, should it not be in the docstring and there be a `add_submodule` method?